client/me/security-checkup/index.jsx

```diff
@@ -0,0 +1,417 @@
+					},
+				}
+			);
+		}
+
+		return (
+			<SecurityCheckupNavigationItem
+				path={ '/me/security/account-recovery' }
+				materialIcon={ icon }
+				text={ translate( 'Recovery Email' ) }
+				description={ description }
+			/>
+		);
+	}
+
+	renderRecoveryPhone() {
+		const {
+			accountRecoveryPhone,
+			accountRecoveryPhoneActionInProgress,
+			accountRecoveryPhoneValidated,
+			translate,
+		} = this.props;
+
+		if ( accountRecoveryPhoneActionInProgress ) {
+			return this.renderPlaceholderNavItem();
+		}
+
+		let icon, description;
+
+		if ( ! accountRecoveryPhone ) {
+			icon = this.getWarningIcon();
+			description = translate( 'You do not have a recovery SMS number.' );
+		} else if ( ! accountRecoveryPhoneValidated ) {
+			icon = this.getWarningIcon();
+			description = translate(
+				'You still need to validate your recovery SMS number: {{strong}}%(recoveryPhoneNumber)s{{/strong}}',
+				{
+					args: {
+						recoveryPhoneNumber: accountRecoveryPhone.numberFull,
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+		} else {
+			icon = this.getOKIcon();
+			description = translate(
+				'You have set {{strong}}%(recoveryPhoneNumber)s{{/strong}} as your recovery SMS number.',
+				{
+					args: {
+						recoveryPhoneNumber: accountRecoveryPhone,
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+		}
+
+		return (
+			<SecurityCheckupNavigationItem
+				path={ '/me/security/account-recovery' }
+				materialIcon={ icon }
+				text={ translate( 'Recovery SMS Number' ) }
+				description={ description }
+			/>
+		);
+	}
+
+	renderTwoFactorAuthentication() {
+		const { translate, userSettings } = this.props;
+
+		if ( ! userSettings.initialized || userSettings.fetchingData ) {
+			return this.renderPlaceholderNavItem();
+		}
+
+		let icon, description;
+		if ( userSettings.settings.two_step_enabled ) {
+			icon = this.getOKIcon();
+			description = translate(
+				'You have two-step authentication {{strong}}enabled{{/strong}} using an app.',
+				{
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+		} else if ( userSettings.settings.two_step_sms_enabled ) {
+			icon = this.getOKIcon();
+			description = translate(
+				'You have two-step authentication {{strong}}enabled{{/strong}} using SMS messages to {{strong}}%(phoneNumber)s{{/strong}}.',
+				{
+					args: {
+						phoneNumber: userSettings.settings.two_step_sms_phone_number,
+					},
+					components: {
+						strong: <strong />,
+					},
+				}
+			);
+		} else {
+			icon = this.getWarningIcon();
+			description = translate( 'You do not have two-step authentication enabled.' );
+		}
+
+		return (
+			<SecurityCheckupNavigationItem
+				path={ '/me/security/two-step' }
+				materialIcon={ icon }
+				text={ translate( 'Two-Step Authentication' ) }
+				description={ description }
+			/>
+		);
+	}
+
+	renderVerticalNav() {
+		return (
+			<VerticalNav>
+				{ this.renderAccountEmail() }
```

Comment:
It might be already on your radar, but I'd suggest splitting each of these rows/`render*` calls into separate components. On their own, they are not that complicated, but even this "skeleton" version of the component is already over 400 lines long and I imagine it will get longer as more logic is added.

Reply:
That was definitely starting to become more apparent as I added new rows -- the code _and_ the dependencies started to become unwieldy. So I'll shift the new rows to components tomorrow and then loop back to the items I've already implemented.